src/site/content/en/blog/aurora-resource-inlining/index.md

```diff
@@ -0,0 +1,212 @@
+
+### Font inlining is now available in Next.js and Angular
+
+When framework developers implement optimization in the underlying tooling, they make it easier for
+existing and new applications to enable it, bringing the improvement to the entire ecosystem.
+
+This improvement is enabled by default from Next.js v10.2 and Angular v11. Both have support for
+inlining Google and Adobe fonts. Angular is expecting to introduce the latter in v12.2.
+
+You can find the implementation of [font inlining in Next.js on
+GitHub](https://github.com/vercel/next.js/pull/14746), and learn more about the optimization in the
+context of Angular in this [video](https://www.youtube.com/watch?v=yOpy9UMQG-Y).
+
+## Inlining critical CSS
+
+Another enhancement involves improving the [First Contentful Paint (FCP)](/fcp) and [Largest
+Contentful Paint (LCP)](/lcp) metrics by inlining critical CSS. We define the critical CSS of a page
+as all the styles that are used at its initial rendering. You can read more about the topic in the
+article [Defer non-critical CSS](/defer-non-critical-css/).
+
+We observed that many applications are loading styles synchronously, which blocks application
+rendering. A quick fix is to load the styles asynchronously. Rather than loading the scripts with
+`media="all"`, set the value of the media attribute to print, and once the loading completes,
+replace the attribute value to all:
+
+```html
+<link rel="stylesheet" href="..." media="print" onload="this.media='all'">
+```
+
+This practice, however, can cause flickering of unstyled content.
+
+
+<figure class="w-figcaption">
+  {% Video
+    src=["video/S838B7UEsdXmwrD8q5gvNlWTHHP2/6ZSZwK3Z0bzK90aunxrW.mp4"],
+    autoplay=true,
+    loop=true,
+    muted=true,
+    playsinline=true
+  %}
+  <figcaption style="text-align: center;">
+    The page appears to flicker as the styles load in.
+  </figcaption>
+</figure>
+
+The video above shows the rendering of a page, which loads its styles asynchronously. The flicker
+happens because the browser first starts downloading the styles, then renders the HTML
+which follows. Once the browser downloads the styles, it triggers the `onload` event of the link
+element, updating the `media` attribute to `all`, and applies the styles to the DOM.
+
+During the time between rendering the HTML and applying the styles the page is partially unstyled.
+When the browser uses the styles, we see flickering, which is a bad user experience and results in
+regressions in [Cumulative Layout Shift (CLS)](/cls/).
+
+To improve the loading behavior, together with asynchronous style loading, we use [critical CSS
+inlining](/extract-critical-css/). The [critters](http://npmjs.com/package/critters) tool can find
+which styles are used on the page, by looking at the selectors in a stylesheet and matching them
+against the HTML. When it finds a match, it considers the corresponding styles as part of the
+critical CSS, and inlines them.
+
+Let's look at an example:
+
+```html
```

Comment:
It would be nice to use the compare component for this (I tried to add it with the github editor but it tripped up on the code formatting): https://web.dev/handbook/web-dev-components/#compare-with-custom-labels

So for the first one, wrap it in:

{% Compare 'worse', 'Example before inlining' %}

{% endCompare %}

and the second would be:

{% Compare 'better', 'Example after inlining' %}

{% endCompare %}

Reply:
Just updated the post to use the compare component.